### ISSUE TITLE
Error: Could not poll API: this.tv.getAmbilightPlusHueState is not a function

### ISSUE THREAD

**Issue by Stormbringer11** [foxriver76/ioBroker.philips-tv-android#1]
Hallo Moritz,

erst mal vielen Dank, dass du dich dem Adapter annimmst. Du bist der große Philips Retter ;o)

Die Erste Version, lief entgegen dem bestehenden Adapter super und auf anhieb. Wollte dich eigentlich fragen ob man noch gewisse Funktionen mit einbauen kann und habe vorsichtshalber geupdated damit es zu keinen Wunschüberschneidungen kommt. 

Aber in den aktuellen Version verweigert der Adapter komplett den Dienst. Im Log kommt immer folgender Error und der Adapter bleibt Orange:

Error: Could not poll API: this.tv.getAmbilightPlusHueState is not a function

Könntest du dir das bitte mal anschauen. Mein Fernseher Modell (API Version 6) unterstützt eigentlich Ambilight und Hue. Mit dem Mic Script funktioniert es noch einwandfrei. 

Node.js v16.16.0
NPM  8.15.1
in einer VM mit Ubuntu 22

Danke :o)

**Comment by foxriver76** [comment]
Danke für dein Feedback, freut mich sehr. 

Ich vermute nicht jeder TV unterstützt zwingend diesen Endpunkt und habe eine entsprechende Logik eben gepushed. Der State wird nur noch erstellt, wenn der Endpunkt verfügbar ist und dann auch nicht mehr abgefragt. Gerne ausprobieren - auf Feature Wünsche bin ich gespannt. 

beste Grüße

**Comment by foxriver76** [comment]
Andere Idee, ich pinne den Commit der Lib, das sollte gehen.

**Comment by Stormbringer11** [comment]
Perfekt wird wieder grün ;o) Cool, Ambi und Hue Funktion geht. Aber Ambilight aus/an schalten funktioniert über Adapter nicht, nur über Mic Skript.

**Comment by foxriver76** [comment]
Ach so.. das ist kein Ambilight an aus, das ist nur ein Ambilight + Hue synchronisieren an/aus.

**Comment by foxriver76** [comment]
Ambilight an aus kann ich bei Gelegenheit am Sonntag oder Montag einbauen.

**Comment by Stormbringer11** [comment]
Klar, das wär auch einer meiner Wünsche gewesen, dass man die verschiedenen Ambilight Modis auswählen kann wie im Skript.

Der Link wäre 
https://github.com/Mic-M/iobroker-philips-tv-script

Das wären die Funktionen dann:
`pCommands['Cmd: Ambilight Off']             = ['ambilight/power',   '{"power":"Off"}'];
pCommands['Cmd: Ambilight On']              = ['ambilight/power',   '{"power":"On"}'];
pCommands['Cmd: Ambilight On Video Standard'] = ['ambilight/currentconfiguration',   '{"styleName":"FOLLOW_VIDEO","isExpert":false,"menuSetting":"STANDARD"}'];
pCommands['Cmd: Ambilight On Video Immersive'] = ['ambilight/currentconfiguration',   '{"styleName":"FOLLOW_VIDEO","isExpert":false,"menuSetting":"IMMERSIVE"}'];
pCommands['Cmd: Ambilight On Video Natural'] = ['ambilight/currentconfiguration',   '{"styleName":"FOLLOW_VIDEO","isExpert":false,"menuSetting":"NATURAL"}'];
pCommands['Cmd: Ambilight On Video Vivid'] = ['ambilight/currentconfiguration',   '{"styleName":"FOLLOW_VIDEO","isExpert":false,"menuSetting":"VIVID"}'];
pCommands['Cmd: Ambilight On Video Game'] = ['ambilight/currentconfiguration',   '{"styleName":"FOLLOW_VIDEO","isExpert":false,"menuSetting":"GAME"}'];
pCommands['Cmd: Ambilight On Video Comfort'] = ['ambilight/currentconfiguration',   '{"styleName":"FOLLOW_VIDEO","isExpert":false,"menuSetting":"COMFORT"}'];`

Bei mir funktionieren aber beim Adapter alle Funktionen nicht die im Ordner "Keys" sind. Beim Skript gehts schon. Vielleicht könntest dir das auch mal anschauen?

**Comment by Stormbringer11** [comment]
wow du bist ja flott und gut :o)

Ja doch jetzt kommt eine. Bei  PlayPause kommt das z. B.

Could not send key "PlayPause": ESOCKETTIMEDOUT

Dropdown und boolean für on/off (cool wär auch noch mit Stateabfrage vom Gerät) klingt sehr gut. Die Frage ist nur, kriegt man das über Adapter hin, wenn man den Style von Automatik auf Manuell ändert, dass er das jeweils im Dropdown umstellt? Also wenn man Automatik Modus "lebendig" z. B. auswählt und dann manuell "Lava" wählt. Über JSON muss man das ja separat ansteuern auto/manuell. Ich würd die manuellen nur irgendwie besonders kennzeichnen. Wenn der manuelle Modus mal aktiviert ist geht der normale Automatik Ambilight Modus nämlich nicht mehr. Das könnte Anfänger auch verwirren.

**Comment by foxriver76** [comment]
Das mit dem Key Press ist strange, 6er API sollte die Keys unterstützen, eben nochmals bei mir getestet. Habe da ad-hoc keinen Schimmer woran das liegen sollte. 

Ambilight Status ist jetzt auf GitHub drin, schalten geht auch, aktuell macht an `FOLLOW_VIDEO` auf setting `GAME`

ich gehe das mal noch durch bzgl. Implementierung der Styles, auch das mit manuell habe ich bislang nicht auf dem Schirm.

**Comment by Stormbringer11** [comment]
Skript habe ich gerade nochmal getestet, da geht Tastendrücken 🙈🤷🏻‍♂️

Ambilight haut bei mir leider nicht hin. Bei aus öffnet er komischerweise das Ambimenü bei ein macht er gar nichts. Im LOG steht.



philips-tv-android.0 | 2022-08-12 17:15:58.598 | debug | Sending key "AmbilightOnOff"
-- | -- | -- | --
philips-tv-android.0 | 2022-08-12 17:15:58.598 | debug | state change philips-tv-android.0.keys.ambilightOnOff: true
philips-tv-android.0 | 2022-08-12 17:15:49.003 | debug | Sending key "AmbilightOnOff"
philips-tv-android.0 | 2022-08-12 17:15:49.003 | debug | state change philips-tv-android.0.keys.ambilightOnOff: false
philips-tv-android.0 | 2022-08-12 17:15:35.549 | debug | Sending key "AmbilightOnOff"
philips-tv-android.0 | 2022-08-12 17:15:35.549 | debug | state change philips-tv-android.0.keys.ambilightOnOff: true
philips-tv-android.0 | 2022-08-12 17:15:26.659 | debug | Sending key "AmbilightOnOff"
philips-tv-android.0 | 2022-08-12 17:15:26.658 | debug | state change philips-tv-android.0.keys.ambilightOnOff: false

Final wird beim boolean state aber nur on off dann gesendet, das ist nur provisorisch oder?

**Comment by foxriver76** [comment]
Du benutzt ja gerade ein key und er wird erfolgreich gesendet. ;-)

Ambilight State ist unter `settings.ambilightActive` alles unter `keys` sind Tasten der Fernbedienung, unter anderem gibt es da den Ambilight Knopf, welcher in der API `AmbilightOnOff` heißt

**Comment by Stormbringer11** [comment]
Ok,  das ist ein gutes Argument 🙈😂 Sorry geht wenn man den richtigen DP nimmt 😉 Aber wenn das andere ein Key ist und er das annimmt, wird’s noch stranger warum er bei den anderen den esockettimeout hat?

**Comment by foxriver76** [comment]
Es gibt nun fürs erste einen custom ambilight command state in settings Kanal, da kann man jegliches Kommando hin senden. Das User freundlichere schaue ich mir mal noch an. Kannst du ein Muster bei den Tasten fest stellen? Sind es jedes mal wieder die gleichen, welche nicht funktionieren?

**Comment by Stormbringer11** [comment]
Habe es gerade nochmal probiert mit den Tasten. Das Muster sind weniger die Tastenarten sondern anscheinend die Eingänge, was irgendwie gar keinen Sinn ergibt oder? Wobei sie jetzt größtenteils funktionieren. Nur einmal bei Harmony Aktionswechsel auf Apple TV kamen wieder Fehler 

philips-tv-android.0 | 2022-08-13 14:25:26.835 | error | Could not change mute status: socket hang up
philips-tv-android.0 | 2022-08-13 14:25:00.391 | error | Could not send key "Home": ESOCKETTIMEDOUT

und sogar, dass der TV aus ist (was er aber nicht war)

philips-tv-android.0 | 2022-08-13 14:23:22.926 | debug | The TV seems to be turned off

Aber wenn sonst keiner das Problem hat musst du wegen mir dich da nicht rumstressen. Hab den Harmony Adapter, brauch die Tasten nicht. Bin happy mit den Ambilight spielereien dann ;D
